### ISSUE TITLE
Shift-tab does not close Picker popup

### ISSUE THREAD

**Issue by mikelambert** [airbnb/react-dates#658]
I am using a `<DateRangePicker>`. Tabbing out of the End field (to the right) works well. Unfortunately, shift-tabbing out of the Start (to the left) has issues. The focus caret is moved to the previous `<input>` outside of the `DateRangePicker`, but the `DateRangePicker__picker` pop-up stays open. It seems that neither `onFocusChange(null)` nor `onClose` are called when I shift-tab out of the Start field.

**Comment by timhwang21** [comment]
Hm, are you sure about that? On Storybook, it seems to work correctly.

http://www.giphy.com/gifs/l41JMVOpsFnIIfwHe

**Comment by mikelambert** [comment]
I was sure about it in my setup, but my setup may have conflating factors, and definitely can see your GIF. Let me look into it more.

I'm attempting to emulate the airbnb approach, with an 'anytime'-like div shows up with no dates selected. It's possible I've screwed things up in building that up myself, since it doesn't appear to be built-in.

**Comment by mikelambert** [comment]
Sorry, seems to have been unique to my setup! I spent some time debugging this, sorry I didn't spend more, but glad you had a Storybook that let you prove things easily.

In case anyone else cares what my case was:

The callbacks were sent out as `onFocusChange(null)` and then `onClose`. I'm using controlled components, so my `onClose` called `setState({focused: false})`, which triggered a re-render. The re-render was re-ordering my components (ie, trying to make the 'anytime' div appear above/below the `DateRangePicker`), which was causing it to be reinitialized, and mistakenly re-set-up with the initial `{focusedInput: 'startDate'}`.

In my case, I switched it to instead keep the same divs, and instead just use `{visibility: 'hidden'}` to show the appropriate div. Then now updating divs, instead of moving/recreating them, and everything seems to work as well as it does in your Storybook. Thanks!
